tests/draft2019-09/optional/format/time.json

```
@@ -9,15 +9,60 @@
                 "valid": true
             },
             {
-                "description": "a valid time string with leap second",
+                "description": "a valid time string with leap second, Zulu",
                 "data": "23:59:60Z",
                 "valid": true
             },
             {
-                "description": "a valid time string with leap second with offset",
+                "description": "invalid leap second, Zulu (wrong hour)",
+                "data": "22:59:60Z",
+                "valid": false
+            },
+            {
+                "description": "invalid leap second, Zulu (wrong minute)",
+                "data": "23:58:60Z",
+                "valid": false
+            },
+            {
+                "description": "valid leap second, positive time-offset",
+                "data": "01:29:60+01:30",
+                "valid": true
+            },
+            {
+                "description": "valid leap second, large positive time-offset",
+                "data": "23:29:60+23:30",
+                "valid": true
+            },
```

@karenetheridge What do you think about the unknown local offset case via https://tools.ietf.org/html/rfc3339#section-4.3 ? Given there would be atleast one local time where the leap second would be permitted. For example:

- `01:29:60-00:00` which would be valid under a local time of `01:30`.

```suggestion
            },
            {
                "description": "valid leap second due to unknown local offset",
                "data": "01:29:60-00:00",
                "valid": true
            },
```

---

If taking the above, perhaps it would be good to also contain test that `+00:00` doesn't beave the same way, and that  `+00:00` would behave like `Z`.

---

I don't think that's what the RFC is saying. If `-00:00` is used, then the time in UTC is the same -- it's just indicating that UTC is not the preferred reference point, but we don't know what the preferred reference point is. The time `01:29:60-00:00` is the same point in time as `01:29:60Z`, which is not a valid time for a leap second.

---

I'll add some tests for `+00:00`.

---

I’ve re-read that part of RFC3339 (and the clearer explanation in RFC2822) and you’re totally right. Thanks for clarifying!